### ISSUE TITLE
[2.5.0] Markdown parsing failed

### ISSUE THREAD

**Issue by zombieJ** [typescript-eslint/typescript-eslint#1123]
**What code were you trying to parse?**

Markdown text:

``````md
There are `primary` button, `default` button, `dashed` button, `danger` button and `link` button in antd.

```jsx
import { Button } from 'antd';

ReactDOM.render(
  <div>
    <Button type="primary">Primary</Button>
    <Button>Default</Button>
    <Button type="dashed">Dashed</Button>
    <Button type="danger">Danger</Button>
    <Button type="link">Link</Button>
  </div>,
  mountNode,
);
```

``````

**What did you expect to happen?**

Works eslint normal as `2.4.0`

**What actually happened?**

> error  Parsing error: '>' expected

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.5.0` |
| `TypeScript`                | `3.6.4` |
| `ESLint`                    | `6.5.1` |
| `node`                      | `10.16.3` |
| `npm`                       | `6.9.0` |

**Comment by bradzacher** [comment]
I don't understand - how are you using our parser to parse markdown?
What config are you using?

**Comment by zombieJ** [comment]
Here is the config file: https://github.com/ant-design/ant-design/blob/master/.eslintrc.js

**Comment by bradzacher** [comment]
It will probably work if you set this in your .eslintrc.js
```json
parserOptions: {
  ecmaFeatures: {
    jsx: true,
  },
}
```

**Comment by zombieJ** [comment]
Thanks for quick response. Tested, seems still meet same error.
<img width="385" alt="截屏2019-10-22下午1 51 55" src="https://user-images.githubusercontent.com/5378891/67261413-2191f080-f4d3-11e9-8375-012bf4939990.png">

<img width="689" alt="截屏2019-10-22下午1 52 24" src="https://user-images.githubusercontent.com/5378891/67261453-48e8bd80-f4d3-11e9-852e-252f74c9d708.png">

**Comment by JounQin** [comment]
+1, my [eslint-mdx](https://github.com/rx-ts/eslint-mdx) breaks due to this, and I'vue wasted a whole afternoon to figure out this is a BREAKING CHANGE from `@typescript-eslint/parser`...

**Comment by bradzacher** [comment]
Again, as per #1124 - it is a bug, unintentionally introduced.

I understand you're frustrated, but please understand that ESLint is a very complex, and extensive environment.

ESLint itself calls into our parser in a certain format.
There are parsers which call into our parser in a different format.
And apparently, there are preprocessors which call into our parser in another different format.

TBH, I didn't even know that this particular use case existed before today.

**Comment by JounQin** [comment]
@bradzacher Sorry for my impatient, I was just too upset at that time.

I personally wrote an eslint parser/plugin for `mdx` either, so I do understand the complexity of eslint. There are so many edge cases.
